### DISCUSSION TITLE
Milvus Mishard K8 Cluster pods not running, They stay in Init:0/1 and Init:0/2 states.

### DISCUSSION THREAD

**Discussion by burakakkas** [milvus-io/milvus#5422] [Q&A and General discussion]
Hello guys, Im actively working on Milvus-Mishard HA K8 Solutions. Im trying to build Mishard cluster on k8 but I have a problem about pods. My mysql pod is running and connected to persistent volume. But other 3 pods which are mishard, write-only and read-only pods not working. They are stay in Init:0/2 and Init:0/1 status. When I described pods they say "Started container wait-for-mysql"

Here is my current setup with 2 master and 2 worker,
https://paste.ubuntu.com/p/ggfvJ8ZKWH/

Here is my persistence Volume config.yaml,
https://paste.ubuntu.com/p/qqmnZP3kYP/

Here is the pods describe output:
https://paste.ubuntu.com/p/xbtgjznSV2/

I used the this init command while setup cluster,
```sh
sudo kubeadm init --control-plane-endpoint "vipmaster:8443" --upload-certs --pod-network-cidr=10.244.0.0/16
```
Here is my helm command for milvus,
```sh
helm install --set cluster.enabled=true --set persistence.enabled=true --set mysql.enabled=true --set mysql.persistence.accessMode=ReadWriteMany --set mysql.imageTag=5.7.14 my-release .
```

My real network subnet is 192.168.1.0/24 but I used 10.244.0.0/16 for pod network for k8 cluster. Because otherwise my pods ip overlaps with my network machines. And they crash too much

When I exec the container with this command,
```sh
kubectl exec -it my-release-milvus-writable-57b685b459-lkjrg -c wait-for-mysql -- /bin/bash
```
and write this,
```sh
nc -z -w 2 my-release-mysql 3306
```
shell says cannot find my-release-mysql. It cannot point mysql pod I guess.

I used milvus helm chart with default configs. I did not edit values.conf. I use command line parameter while exewcuting helm install command.

I also try to build mishard on ubuntu server with just 1 master node. It worked with same init settings. So problem is about multi node architecture and mishard configs maybe. I cannot figure out problem because describe logs dont give any error.

**Comment by shiyu22** [comment]
I think the question about this has already been answered in [slack].(https://milvusio.slack.com/archives/C012DUVUFNZ).
